SpreadCheetah/Styling/NumberFormat.cs

```diff
@@ -0,0 +1,93 @@
+using SpreadCheetah.Helpers;
+using System.Globalization;
+
+namespace SpreadCheetah.Styling
+{
+    /// <summary>
+    /// Format that defines how a number or <see cref="DateTime"/> cell should be displayed.
+    /// May be a custom format, initialised by <see cref="NumberFormat.Custom"/>, or a standard format, initialised by <see cref="NumberFormat.Standard"/>
+    /// </summary>
+#pragma warning disable CA1815, CA2231 // Override equals and operator equals on value types - equals operator not appropriate for this type
+    public readonly struct NumberFormat : IEquatable<NumberFormat>
```

Comment:
I think we can make this a `readonly record struct` so that the compiler would generate the equality and the `GetHashCode` methods.

Reply:
Good call, yes.